frontend/jest.config.ts

```diff
@@ -20,7 +20,7 @@ const config: Config = {
 
   // Indicates whether the coverage information should be collected while executing the test
   collectCoverage: true,
-  testEnvironment: 'jsdom',
+  testEnvironment: "node",
```

Comment:
we need jsdom to test react code

Reply:
jsdom breaks my code

Reply:
Fixed by adding 

test("TextEncoder is defined", () => {
  expect(() => new TextEncoder()).not.toThrow()
})